backend-frontend/app/Http/Controllers/OpenAIController.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Preference;
+use Illuminate\Http\Request;
+
+class OpenAIController extends Controller
+{
+    public function sendMessage(Request $request) {
```

Comment:
split logic in this method into OpenAIService.php where you'll handle CURL request, sanitization and whatever else you need to do